src/main/java/org/testng/internal/objects/SimpleObjectDispenser.java

```diff
@@ -0,0 +1,41 @@
+package org.testng.internal.objects;
+
+import org.testng.TestNGException;
+import org.testng.internal.InstanceCreator;
+import org.testng.internal.objects.pojo.CreationAttributes;
+import org.testng.internal.objects.pojo.DetailedAttributes;
+import org.testng.internal.objects.pojo.BasicAttributes;
+
+/**
+ * A plain vanilla Object dispenser
+ */
+class SimpleObjectDispenser implements IObjectDispenser {
+
+  @Override
+  public void setNextDispenser(IObjectDispenser dispenser) {
+    //We are not going to be doing anything with this downstream dispenser since we are
+    //last in the chain of responsibility.
+  }
+
+  @Override
+  public Object dispense(CreationAttributes attributes) {
+    DetailedAttributes detailed = attributes.getDetailedAttributes();
+    if (detailed != null) {
+      return InstanceCreator.createInstance(detailed.getDeclaringClass(),
+          detailed.getClasses(), detailed.getXmlTest(), detailed.getFinder(), detailed.getFactory(),
+          detailed.isCreate(), detailed.getErrorMsgPrefix());
+    }
+    BasicAttributes basic = attributes.getBasicAttributes();
+    if (basic == null) {
+      throw new TestNGException("Encountered problems in creating instances.");
+    }
+    if (basic.getRawClass() == null) {
+      try {
+        return InstanceCreator.newInstance(basic.getTestClass().getRealClass());
+      } catch (TestNGException e) {
```

Comment:
Why not removing the catch here?

Reply:
Some tests tend to try instantiating using a unsupported way, which causes the `newInstance()` to throw exception. In those cases, we are just trying to find out using which instantiation mechanism the object creation can happen. I need to dig in a bit more deeper on this to find out what is going on.